Makefile

```diff
@@ -62,7 +79,7 @@ ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 test: manifests generate fmt vet ## Run tests.
 	mkdir -p ${ENVTEST_ASSETS_DIR}
 	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
-	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out; make -C pkg/eraser/ run
+	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test $(shell pwd)/pkg/.. $(shell pwd)/controllers/.. -coverprofile cover.out; make -C pkg/eraser/ run
```

Comment:
looking at the logs, looks like we are still only testing pkg/eraser here

in v0.9, controller-runtime moved away from setup-envtest.sh script to a binary

i think we can replace this to make it simpler:
```
test: manifests generate fmt vet envtest ## Run tests.
	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out; make -C pkg/eraser/ run
```

and then add after kustomize
```
ENVTEST = $(shell pwd)/bin/setup-envtest
.PHONY: envtest
envtest: ## Download envtest-setup locally if necessary.
	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
```

Reply:
I was planning to go through the code base and change a few things, thanks for starting this.

Reply:
Here is the log now:
```
KUBEBUILDER_ASSETS="/home/runner/.local/share/kubebuilder-envtest/k8s/1.[23](https://github.com/Azure/eraser/runs/5165091614?check_suite_focus=true#step:5:23).3-linux-amd64" go test ./... -coverprofile cover.out; make -C pkg/eraser/ run
?   	github.com/Azure/eraser	[no test files]
?   	github.com/Azure/eraser/api/v1alpha1	[no test files]
ok  	github.com/Azure/eraser/controllers	5.973s	coverage: 20.0% of statements
?   	github.com/Azure/eraser/controllers/imagejob	[no test files]
?   	github.com/Azure/eraser/controllers/imagelist	[no test files]
make[1]: Entering directory '/home/runner/work/eraser/eraser/pkg/eraser'
cd /home/runner/work/eraser/eraser/pkg/eraser; go test . -coverprofile cover.out
ok  	github.com/Azure/eraser/pkg/eraser	0.013s	coverage: 15.3% of statements
```